extensions/wikia/ThemeDesigner/ThemeSettingsPersistence.php

```diff
@@ -0,0 +1,197 @@
+<?php
+
+class ThemeSettingsPersistence {
+	/** @var ThemeSettings $themeSettings */
+	private $themeSettings;
+	/** @var int $cityId */
+	private $cityId;
+
+	public function __construct( ThemeSettings $themeSettings ) {
+		$this->themeSettings = $themeSettings;
+		$this->cityId = $themeSettings->getCityId();
+	}
+
+	public function getSettings() {
+		return WikiFactory::getVarValueByName(
+			ThemeSettings::WikiFactorySettings,
+			$this->cityId
+		);
+	}
+
+	public function saveSettings( $settings ) {
+		global $wgUser;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)